internal/controller/serverbios_controller.go

```diff
@@ -0,0 +1,838 @@
+	}
+	log.V(1).Info("Created serverMaintenance", "serverMaintenance", serverMaintenance.Name, "serverMaintenance label", serverMaintenance.Labels, "Operation", opResult)
+
+	err = r.patchMaintenanceRequestRefOnServerBIOS(ctx, log, serverBIOS, serverMaintenance)
+	if err != nil {
+		return false, fmt.Errorf("failed to patch serverMaintenance ref in serverBIOS status: %w", err)
+	}
+
+	log.V(1).Info("Patched serverMaintenance on serverBIOS")
+
+	return true, nil
+}
+
+func (r *ServerBIOSReconciler) getReferredServer(
+	ctx context.Context,
+	log logr.Logger,
+	serverRef *corev1.LocalObjectReference,
+) (*metalv1alpha1.Server, error) {
+	key := client.ObjectKey{Name: serverRef.Name}
+	server := &metalv1alpha1.Server{}
+	if err := r.Get(ctx, key, server); err != nil {
+		log.V(1).Error(err, "failed to get referred server")
+		return server, err
+	}
+	return server, nil
+}
+
+func (r *ServerBIOSReconciler) getReferredServerMaintenance(
+	ctx context.Context,
+	log logr.Logger,
+	serverMaintenanceRef *corev1.ObjectReference,
+) (*metalv1alpha1.ServerMaintenance, error) {
+	key := client.ObjectKey{Name: serverMaintenanceRef.Name, Namespace: r.ManagerNamespace}
+	serverMaintenance := &metalv1alpha1.ServerMaintenance{}
+	if err := r.Get(ctx, key, serverMaintenance); err != nil {
+		log.V(1).Error(err, "failed to get referred serverMaintenance obj")
+		return serverMaintenance, err
+	}
+
+	return serverMaintenance, nil
+}
+
+func (r *ServerBIOSReconciler) getReferredserverBIOS(
+	ctx context.Context,
+	log logr.Logger,
+	referredBIOSSetteingRef *corev1.LocalObjectReference,
+) (*metalv1alpha1.ServerBIOS, error) {
+	key := client.ObjectKey{Name: referredBIOSSetteingRef.Name, Namespace: metav1.NamespaceNone}
+	serverBIOS := &metalv1alpha1.ServerBIOS{}
+	if err := r.Get(ctx, key, serverBIOS); err != nil {
+		log.V(1).Error(err, "failed to get referred BIOSSetting")
+		return serverBIOS, err
+	}
+	return serverBIOS, nil
+}
+
+func (r *ServerBIOSReconciler) patchServerBIOSRefOnServer(
+	ctx context.Context,
+	log logr.Logger,
+	server *metalv1alpha1.Server,
+	serverBIOSReference *corev1.LocalObjectReference,
+) error {
+	if server.Spec.BIOSSettingsRef == serverBIOSReference {
+		return nil
+	}
+
+	var err error
+	serverBase := server.DeepCopy()
+	server.Spec.BIOSSettingsRef = serverBIOSReference
+	if err = r.Patch(ctx, server, client.MergeFrom(serverBase)); err != nil {
+		log.V(1).Error(err, "failed to patch bios settings ref")
+		return err
+	}
+	return err
+}
+
+func (r *ServerBIOSReconciler) patchMaintenanceRequestRefOnServerBIOS(
+	ctx context.Context,
+	log logr.Logger,
+	serverBIOS *metalv1alpha1.ServerBIOS,
+	serverMaintenance *metalv1alpha1.ServerMaintenance,
+) error {
+	serverBIOSBase := serverBIOS.DeepCopy()
+
+	if serverMaintenance == nil {
+		serverBIOS.Spec.ServerMaintenanceRef = nil
+	} else {
+		serverBIOS.Spec.ServerMaintenanceRef = &corev1.ObjectReference{
+			APIVersion: "metal.ironcore.dev/v1alpha1",
```

Comment:
`serverMaintenance.GroupVersionKind().GroupVersion().String()` should avoid hard-coding the apiVersion.